libyul/Object.cpp

```diff
@@ -69,7 +69,8 @@ set<YulString> Object::qualifiedDataNames() const
 	for (shared_ptr<ObjectNode> const& subObjectNode: subObjects)
 	{
 		yulAssert(qualifiedNames.count(subObjectNode->name) == 0, "");
-		qualifiedNames.insert(subObjectNode->name);
+		if (subObjectNode->name.str() != metadataName() || !dynamic_cast<Data const*>(subObjectNode.get()))
```

Comment:
Hmm, this `!Data` is a bug fix? Do we have a test case for that? There was some bug about trying to refer data objects in files which have empty code (and as such are not compiled).